### PR TITLE
Add tcsh support to --shell

### PR DESCRIPTION
**Pull request by eedeebee**
Signed-off-by: Eric Bloch <eric.bloch@email.com>

### CODE REVIEW

File: commands/env.go [hunk 1]
```diff
@@ -182,6 +186,10 @@ func shellCfgUnset(c CommandLine, api libmachine.API) (*ShellConfig, error) {
 		shellCfg.Prefix = "(setenv \""
 		shellCfg.Suffix = ")\n"
 		shellCfg.Delimiter = "\" nil"
+	case "tcsh":
+		shellCfg.Prefix = "setenv "
```
Comment (drewish) [inline commands/env.go]: Any reason you didn't go with `unsetenv` instead? Downside of the current approach is it leaves envs around just blank:
```
% eval `./docker-machine env -u`
% setenv | grep DOCK
DOCKER_HOST=
DOCKER_TLS_VERIFY=
DOCKER_CERT_PATH=
DOCKER_MACHINE_NAME=
```

File: docs/reference/env.md [hunk 2]
```diff
@@ -25,7 +25,7 @@ a particular machine.
     Options:
 
        --swarm	Display the Swarm config instead of the Docker daemon
-       --shell 	Force environment to be configured for a specified shell: [fish, cmd, powershell], default is sh/bash
+       --shell 	Force environment to be configured for a specified shell: [fish, cmd, powershell, tcsh], default is sh/bash
```
Comment (drewish) [inline docs/reference/env.md]: should `emacs` be in here too?

Comment (eedeebee) [inline docs/reference/env.md]: I found it this way and I honestly wasn't sure why whomever added emacs didn't put it here; was thinking perhaps it wasn't supported code or something so didn't think it was my place to add it.

File: commands/env.go (hunk 1 above)
Comment (eedeebee) [inline commands/env.go]: You're right, I think unsetenv would be better.  I'll try that when I have a moment.

**Comment by drewish** [comment]
Thanks, this is working well for me. Looking forward to it getting merged.

File: docs/reference/env.md (hunk 2 above)
Comment (nathanleclaire) [inline docs/reference/env.md]: It probably just got lost in the mix -- we'd be happy to merge a PR for it :)

**Comment by nathanleclaire** [comment]
LGTM

**Comment by nathanleclaire** [comment]
Thanks @eedeebee !

**Pull request merged**